packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -581,4 +594,74 @@ class EmbraceManager: NSObject {
 
         resolve(true)
     }
+    
+    @objc
+    func logNetworkRequest(
+        _ url: String,
+        httpMethod: String,
+        startInMillis: Double,
+        endInMillis: Double,
+        bytesSent: Double,
+        bytesReceived: Double,
+        statusCode: Double,
+        error: String,
+        resolver resolve: RCTPromiseResolveBlock,
+        rejecter reject: RCTPromiseRejectBlock
+    ) {
+        if Embrace.client == nil {
+            reject("RECORD_LOG_NETWORK_REQUEST_ERROR", "Error recording span, Embrace SDK may not be initialized", nil)
+            return
+        }
+
+        let attributes = attributeStringsFrom(dict: [
+            "http.response.status_code": String(Int(statusCode)),
+            "http.request.body.size": String(Int(bytesSent)),
+            "http.response.body.size": String(Int(bytesReceived)),
+            "http.request.method": httpMethod.uppercased(),
+            "url.full": url,
+        ]);
+
+        Embrace.client?.recordCompletedSpan(name: createNetworkSpanName(url: url, httpMethod: httpMethod),
+                                            type: SpanType.networkRequest, parent: nil,
+                                            startTime: dateFrom(ms: startInMillis), endTime: dateFrom(ms: endInMillis),
+                                            attributes: attributes,
+                                            events: eventsFrom(array: []),
+                                            errorCode: errorCodeFrom(str: error))
+
+        resolve(true)
+    }
+   
+    @objc
+    func logNetworkClientError(
+        _ url: String,
+        httpMethod: String,
+        startInMillis: Double,
+        endInMillis: Double,
+        errorType: String,
+        error: String,
+        resolver resolve: RCTPromiseResolveBlock,
+        rejecter reject: RCTPromiseRejectBlock
+    ) {
+        if Embrace.client == nil {
+            reject("RECORD_LOG_NETWORK_CLIENT_ERROR_ERROR", "Error recording span, Embrace SDK may not be initialized", nil)
+            return
+        }
+        
+        let attributes = attributeStringsFrom(dict: [
+            "http.request.method": httpMethod.uppercased(),
+            "url.full": url,
+            "error.message": error,
+            "error.type": errorType,
+        ]);
+
+        Embrace.client?.recordCompletedSpan(name: createNetworkSpanName(url: url, httpMethod: httpMethod),
+                                            type: SpanType.networkRequest,
+                                            parent: nil,
+                                            startTime: dateFrom(ms: startInMillis), endTime: dateFrom(ms: endInMillis),
+                                            attributes: attributes,
+                                            events: eventsFrom(array: []),
+                                            errorCode: errorCodeFrom(str: error))
```

Comment:
this requires the string to match a particular enum, probably we want to explicitly set it as `errorCode: .failure` or whatever Android is setting

Reply:
done